### ISSUE TITLE
Navigation URLs being resolved only for root

### ISSUE THREAD

**Issue by glnarayanan** [Kikobeats/uno-zen#51]
Hi ! 

First of all, thanks a lot for the lovely template and the excellent work you've done with it.

Coming to the gist, I'm very noobish when it comes to JS, so please correct me if I'm doing something wrong. The issue is as follows:

Whatever URLs I add to the Navigation list, they're resolving to root URL

Ex: 

(Assume root is example.com)

Say I'm adding,

About: example.com/about
Google: google.com

Now, on the home page, two buttons, About, Google are created. But both of them point only to example.com, not to their respective URLs.

I tried using {{log url}} within the {{for each}} blog in links.hbs. The logged output looks good. (/about, google.com).

So I'm wondering where it goes wrong. 

Live site: gln.me

**Comment by ayon115** [comment]
I am also having exactly the same problem. Any workaround ?

**Comment by jcdenton** [comment]
I was experiencing the same issue.
Looks like the builtin Ghost `{{url}}` helper overrides `{{url}}` attribute even inside the `{{#foreach @blog.navigation}}` loop. 
One way to solve this is to rename `links.hbs` into `navigation.hbs` and include it with `{{navigation}}` helper instead of `{{> links}}` and loop over just `navigation` instead of `@blog.navigation`. 
However there is also a more elegant fix: just use `{{this.url}}` instead of `{{url}}` - https://github.com/jcdenton/uno-zen/commit/3ec11ecfce4ca9f8045fab7c842fe6148054f9b8.

**Comment by ayon115** [comment]
I managed to fix the issue. I removed the foreach loop in links.hbs and added the page links like the previous versions of this theme. My links.hbs looks like this now.

`<nav class="navigation left">
  <ul class="links">
    <li class="links item">
      <a href="/" class="link-item" title="" id="tags-button">Home</a>
      <a href="/#open" class="link-item" title="link to {{@blog.title}} blog" id="blog-button">Blog</a>
      <a href="/about" class="link-item" title="" id="tags-button">About</a>
      <a href="/contact" class="link-item" title="" id="tags-button">Contact</a>
    </li>
  </ul>
</nav>`

**Comment by glnarayanan** [comment]
@ayon115 Thought that would definitely work, that is not a solution after all :) 

@jcdenton What you've told sounds probably, but as I've mentioned, I logged the `url` INSIDE the `{{for each}}` block and the logged url appeared proper. I just used `{{log url}}`. Since the logged `url` was proper, I wonder where it gets messed up when passed to `href`.

**Comment by jcdenton** [comment]
@glnarayanan and does my explanation about the attribute being overridden by the builtin `{{url}}` helper contradict the fact that `{{log url}}` would work as expected? BTW using `{{this.url}}` just confirms this.

**Comment by Kikobeats** [comment]
Thank you for report this. Then exists a real solution? Somebody can create a PR with the solution?

I want to use the native Ghost links, but yes, with whatever kind of link.

**Comment by ayon115** [comment]
@glnarayanan Well, may not be a good solution, but it's working for me. Here's my steps to make it work in my website.
1. Edit links.hbs like my earlier message.
2. Add appropriate in Navigation in Ghost Settings panel. (Ex. http://base_url/about)
3. Create static page content with a post from Ghost Settings panel. (Ex- about me page contents)
4. Create page in theme's root directory with name like page-pagename.hbs (Ex - page-about.hbs).
   Such a page of mine looks like this 

```
{{!< default}}
{{#post}}
  <article class="{{post_class}}">
    <header>
      <h1 id="post-title">About me</h1>
    </header>
    <section class="{{post_class}}">
      {{content}}
    </section>
  </article>
{{/post}}
```

Then finally Restart Ghost. It should work.

**Comment by Kikobeats** [comment]
@ayon115 absolutely works setup manually the markup, but the idea is integrate as soon as possible the themes with the Ghost API.

BTW, related: http://themes.ghost.org/docs/navigation

the current code generator of the themes is:

```
{{#foreach @blog.navigation}}
        <a href="{{url}}" class="link-item" id="{{slug}}-button">{{label}}</a>
{{/foreach}}
```

I think that could be enough using `{{url absolute="true"}}` . Let's try

**Comment by jcdenton** [comment]
@Kikobeats I've tried it already, didn't work for me.
The http://themes.ghost.org/docs/navigation doc assumes that 1) the navigation template is called `navigation.hbs` and 2) it is included with `{{navigation}}`, not `{{> smth}}`. 
As we have `links.hbs`, our `{{url}}` works as the default helper http://themes.ghost.org/docs/url so it always returns the _current_ URL for the _current_ scope ("/" for the index page).

**Comment by Kikobeats** [comment]
Fixed in #54 and available in `2.3.5` tag.

Notes that all external links of the blog are opened with `target=_blank`!
